src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.CompletionSymbolDisplay.cs

```diff
@@ -0,0 +1,127 @@
+                {
+                    builder.Append(symbol.Name);
+                }
+
+                if (symbol.Parameters.Length > 0)
+                {
+                    builder.Append('[');
+                    AddParameters(symbol.Parameters, builder);
+                    builder.Append(']');
+                }
+
+                return builder.ToString();
+            }
+
+            private static string ToDisplayString(IMethodSymbol symbol)
+            {
+                using var _ = PooledStringBuilder.GetInstance(out var builder);
+                switch (symbol.MethodKind)
+                {
+                    case MethodKind.Ordinary:
+                        builder.Append(symbol.Name);
+                        break;
+                    case MethodKind.UserDefinedOperator:
+                    case MethodKind.BuiltinOperator:
+                        builder.Append("operator ");
+                        builder.Append(SyntaxFacts.GetText(SyntaxFacts.GetOperatorKind(symbol.MetadataName)));
+                        break;
+                    case MethodKind.Conversion:
+                        builder.Append("operator ");
+                        AddType(symbol.ReturnType, builder);
+                        break;
+                }
+
+                AddTypeArguments(symbol, builder);
+                builder.Append('(');
+                AddParameters(symbol.Parameters, builder);
+                builder.Append(')');
+                return builder.ToString();
+            }
+
+            private static void AddParameters(ImmutableArray<IParameterSymbol> parameters, StringBuilder builder)
+            {
+                var first = true;
+                foreach (var parameter in parameters)
+                {
+                    if (!first)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    first = false;
+                    builder.Append(parameter.RefKind switch
+                    {
+                        RefKind.Out => "out ",
+                        RefKind.Ref => "ref ",
+                        RefKind.In => "in ",
+                        _ => ""
+                    });
+
+                    AddType(parameter.Type, builder);
+                    builder.Append($" {parameter.Name}");
+                }
+            }
+
+            private static void AddTypeArguments(IMethodSymbol symbol, StringBuilder builder)
+            {
+                if (symbol.TypeArguments.Length > 0)
+                {
+                    builder.Append('<');
+
+                    var first = true;
+                    foreach (var typeArgument in symbol.TypeArguments)
+                    {
+                        if (!first)
+                        {
+                            builder.Append(", ");
+                        }
+
+                        first = false;
+                        builder.Append(typeArgument.Name);
+                    }
```

Comment:
i would be 100% ok with a stringbuilder extension that takes a TEnumerable and a separator and does this logic :)

Reply:
@CyrusNajmabadi A private extension just for this class, or somewhere else for wider usage?

Reply:
somewhere else :)